src/main/java/org/apache/sling/distribution/servlet/DistributionAgentServlet.java

```diff
@@ -71,7 +73,7 @@ protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse
                 log.debug("distribution response : {}", distributionResponse);
             } catch (Throwable e) {
                 log.error("an unexpected error has occurred", e);
-                ServletJsonUtils.writeJson(response, 503, "an unexpected error has occurred", null);
+                ServletJsonUtils.writeJson(response, 503, e.getMessage(), null);
```

Comment:
I don't think that this statuscode is correct, the service is not unavailable. There was just an (unqualified) exception, and we don't know what exactly happened just based on the exception type. Is it possible to narrow down the exception type and then provide a better matching status code?

Also I don't think if it's a good idea to expose the exception message potentially containing data which should not be exposed.

Reply:
I agree. Statuscode 503 is not suitable. On the other hand this is returning 503 for a long time now. We might break clients if we change it.

Reply:
Agree that the status code is not ideal. However, I think Christian is right. I will track this in a JIRA issue, to be assessed later, since it falls outside of the main purpose of this PR.

Regarding the exception narrowing, even though I agree, the current implementation makes it very hard to do such a thing. If you look at the [code that we are executing](https://github.com/apache/sling-org-apache-sling-distribution-journal/blob/master/src/main/java/org/apache/sling/distribution/journal/impl/publisher/DistributionPublisher.java#L267) it is catching a generic exception and raising this DistributionException, which can contain any type of error message, and we cannot be sure which.

We need to discuss how can we include maybe more metadata in this `DistributionException` which would help an upper module decide the severity and type of error.

Reply:
> Also I don't think if it's a good idea to expose the exception message potentially containing data which should not be exposed.

+1. We can make the error more specific. In this case, I think that ideally the message should be

```
Failed to create content package: Can't distribute package with size greater than 1 Byte, actual 4053
```

Reply:
@cschneider Its seems like implementing the error codes pattern is looking more and more useful. This could be solved by having something like:

```java
public enum Error {
    PACKAGE_SIZE(0, "Package size exceeds the limit threshold."),
    AUTHORIZATION(1, "Not authorized to perform action");
    INTERNAL_ERROR(2, "An error has occured");
   ....
}

public class DistributionException extends Exception {
    public Error errorCode;
}
```

Reply:
Created Jira - [Wrong 503-unavailable status code in Distribution API responses](https://issues.apache.org/jira/browse/SLING-11371)

Reply:
I decided to implement an initial approach to a more structured way of having well-defined external error codes and improve how we propagate the exception metadata. What do you think?

[https://github.com/apache/sling-org-apache-sling-distribution-core/pull/60](https://github.com/apache/sling-org-apache-sling-distribution-core/pull/60)